components/_shared/VLink.vue

```diff
@@ -0,0 +1,120 @@
+<template>
+  <button
+    :class="buttonClasses"
+    :to="link"
+  >
+    <slot></slot>
+  </button>
+</template>
+
+<script>
+export default {
+  props: {
+    typ: String,
+    theme: String,
+    link: String,
+},
+  computed: {
+    buttonClasses() {
+      return ['main-button', this.typ ? `${this.typ}-button` : '', this.theme ? `${this.theme}-button` : ''];
+    }
+  }
+}
+</script>
+
+<style scoped>
+.main-button {
+  background-color: transparent;
+  border: none;
+  color: #2a272d;
+  font-size: 20px;
+  padding: 10px 20px;
+  cursor: pointer;
+}
+
+.primary-button {
+  padding: 10px 20px;
+  background-color: #0E8585;
+  color: white;
+  border: none;
+  border-radius: 10px;
+}
+
+.secondary-button {
+  padding: 10px 20px;
+  background-color: #fff;
+  color: #0E8585;
+  border: none;
+  border-radius: 10px;
+}
+
+</style>
+ChatGPT
+Oto jak możesz przekształcić komponent Vue z kodem, który podałeś, na komponent Vue z TypeScriptem:
+
+vue
+Copy code
+<template>
+  <button
+    :class="buttonClasses"
+    :to="link"
+  >
+    <slot></slot>
+  </button>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue';
+
+enum TypEnum {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+export default defineComponent({
+  props: {
+    typ: {
+      type: String as PropType<keyof typeof TypEnum>,
```

Comment:
`String` type is enough
```suggestion
      type: String,
```